[Sun Xinyu] ip

Echomo-Xinyu opened the pull request:
# Duke
*Duke is a Chatbot that is capable of managing tasks created by the user.*

> “Nowadays, everyone should use some kind of productivity tools to track their progress in numerous tasks.”  - Xinyu ([source](github.com/echomo-Xinyu)

This is,

- command-line interface based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/Echomo-Xinyu/ip/releases/download/v0.1.0-alpha/duke.jar);
2. double-click it;
3. add your tasks;
4. let it manage your tasks for you 😉

And it is **FREE**!

This app includes features such as:

- [X] Managing reminders(todos)
- [X] Managing deadlines
- [X] Managing events
- [ ] more to come!!!

If you are Java programmer, you can use it to practice Java too. Here is the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Echomo-Xinyu committed:
Add basic Greeting, Echoing and Ending

Echomo-Xinyu committed:
Add ability to store input

Echomo-Xinyu committed:
Add Mark as Done function

Echomo-Xinyu committed:
Code Organize

Echomo-Xinyu committed:
Add Support for more tasks

Echomo-Xinyu committed:
add support for Automated Test

Echomo-Xinyu committed:
Add Support for Weird Input

Echomo-Xinyu committed:
Add Support for delete task

Echomo-Xinyu committed:
Improve exception handling

Echomo-Xinyu committed:
Add input file for testing

Echomo-Xinyu committed:
Organize code

Echomo-Xinyu committed:
Add save function

Echomo-Xinyu committed:
Add support for datetime

Echomo-Xinyu committed:
Merge branch 'branch-Level-7' to add save function

Echomo-Xinyu committed:
Merge branch 'branch-Level-8' to add support for datetime

Echomo-Xinyu committed:
Organize file for package

Echomo-Xinyu committed:
Organize code to become more OOP style

Echomo-Xinyu committed:
Add support for loading stored information

Echomo-Xinyu committed:
Optimize logic and fix minor bugs

Echomo-Xinyu committed:
Organize code to better adhere to OOP

Echomo-Xinyu committed:
Add todo for further reference

Echomo-Xinyu committed:
Add Gradle Support

Echomo-Xinyu committed:
Include style check

Echomo-Xinyu committed:
Include basic JUnit test

Echomo-Xinyu committed:
Add Find function

Echomo-Xinyu committed:
Fix coding style

Echomo-Xinyu committed:
Add JavaDoc for files

Echomo-Xinyu committed:
Resolve merge conflict

Echomo-Xinyu committed:
Resolve merge conflict with A-JavaDoc

Echomo-Xinyu committed:
Improve Java Doc

Echomo-Xinyu committed:
Fix style

weiquany commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,130 @@
+     * @param t task object to be modified on
+     * @param b the state the task object is changed to
+     */
+    protected static void taskStateChangePrint(Task t, boolean b) {
```
Perhaps a clearer name can be provided to the boolean `b` ( E.g. isDone) to improve readability.

weiquany commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,130 @@
+    protected static void listPrint(TaskList taskList) {
+        ArrayList<Task> arr = taskList.getTaskArrayList();
+        int count = 1;
+        String result = "Here are the tasks in your list:\n";
```
I noticed that you used `\n` throughout your code, it may be better to replace it with something like `System.lineSeparator()` to ensure that it can work on different OSes.

Jnwkm replied:
Agree with the above comment.

weiquany commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,230 @@
+
+    private void createEvent(TaskEnum taskEnum, String command) throws DukeException {
+        String[] args;
+        Task res;
```
I think the naming `res` may be too vague for this variable that has a large scope. Maybe you can consider changing it, similarly for other parts of the application if you deem that it may suffer from a similar problem.

weiquany commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,130 @@
+
+class Ui {
+    private static final String SEPARATING_LINE = "    ____________________________________________________________";
+    private static final String logo = " ____        _        \n"
```
I think you missed out on capitalizing `logo`. Great job for the rest though!

weiquany reviewed (commented):
Great job overall on adhering to the CS2103 code quality standards! I had a difficult time looking for possible mistakes, and may have made mistakes in my review, do feel free to correct me!

Echomo-Xinyu committed:
Modify variable naming in response to comment

Echomo-Xinyu committed:
Fix style with new CheckStyle plugin

Echomo-Xinyu committed:
Include basic GUI components into Duke

Echomo-Xinyu committed:
Integrate existing logic into GUI

Echomo-Xinyu committed:
Merge branch 'branch-Level-10' to include GUI

Jnwkm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,212 @@
+public class TaskList {
+    private ArrayList<Task> taskArrayList;
+    enum TaskEnum {
+        Todo,
+        Deadline,
+        Event
```
good use of enums here to show the unique types of tasks!

Jnwkm commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,212 @@
+        String description;
+        isDone = !(line.charAt(4) == '0');
+        switch (line.substring(0, 1)) {
+        case "T":
+            description = line.split("\\|", 3)[2];
+            description = description.substring(1);
+            curr = new Todo(description);
+            curr.setDone(isDone);
+            this.addTask(curr);
+            break;
+        case "E":
+            description = line.split("\\|", 4)[2];
+            description = description.substring(1, description.length() - 1);
+            localDate = LocalDate.parse(line.split("\\|", 4)[3].substring(1));
+            curr = new Event(description, localDate);
+            curr.setDone(isDone);
+            this.addTask(curr);
+            break;
+        case "D":
```
good uses of switch cases!

Jnwkm reviewed (commented):
Overall it's very well aligned to the coding practices of Java! Great work!

Echomo-Xinyu committed:
Include Assertion

Assertion is used to ensure certain assumptions hold true upon run.

Echomo-Xinyu committed:
Improve Code Quality

Several Magic Literals have been converted to named variables.
Examples of commands are included to facilitate better understanding of code.

Echomo-Xinyu committed:
Include comment for better understanding

Echomo-Xinyu committed:
Merge pull request #1 from Echomo-Xinyu/branch-A-Assertions

Include Assertion

Echomo-Xinyu committed:
Merge pull request #2 from Echomo-Xinyu/branch-A-CodeQuality

Include comment for better understanding

Echomo-Xinyu committed:
Add Continuous Integration

Echomo-Xinyu committed:
Fix Gradle Set-up after GUI

Echomo-Xinyu committed:
Merge branch 'master' of https://github.com/Echomo-Xinyu/ip

Echomo-Xinyu committed:
Add Support for Help, Greet, End

Functionalies including help, greet, and end have been included.

- An automatic greet has been added at the beginning of the app.
- help keyword will trigger a print of the possible commands.
- after bye keyword, the program will no longer accept any input.

Echomo-Xinyu committed:
Create main.yml

Echomo-Xinyu committed:
Improve JavaDoc for all Java files

All non-private methods are precedented with a JavaDoc.

Previous outdated JavaDoc has been updated accordingly.

Echomo-Xinyu committed:
Merge branch 'master' of https://github.com/Echomo-Xinyu/ip

Echomo-Xinyu committed:
Fix Bug and Include minor improve on GUI

Echomo-Xinyu committed:
Include Ui screenshot

Echomo-Xinyu committed:
Update README as User Guide

Echomo-Xinyu committed:
Finalize features

Echomo-Xinyu committed:
Fix Bug for skip index in delete

Echomo-Xinyu committed:
Fix line break uncompatible issue with Windows

Echomo-Xinyu committed:
Fix Uncompatibility with Windows and Mac

Echomo-Xinyu committed:
Fix uncompatible separator

Echomo-Xinyu committed:
Fix Todo Empty description issue

Echomo-Xinyu committed:
Fix setup issue on different OS
